### PR TITLE
Jetpack Partner Portal: implement UI for creating a site page.

### PR DESCRIPTION
**Pull request by yashwin**
Closes https://github.com/Automattic/jetpack-avalon/issues/42

## Proposed Changes

This PR implements UI for creating a site page.

## Testing Instructions

- Open the Jetpack Cloud live link
- Append the URL with `/partner-portal/create-site` and verify the below screen is shown.

Screenshot 2023-08-29 at 8 18 30 AM

- Verify this screen looks good on other devices

Mobile

mobile (33)

-----

Tablet

mobile (34)

-----

Large screen

mobile (35)

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/card-content.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	storage: string;
+	logo: JSX.Element | null;
+}
+
+export default function CardContent( { planSlug }: { planSlug: string } ) {
+	const translate = useTranslate();
+	const tooltipRef = useRef< HTMLDivElement | null >( null );
+	const [ showPopover, setShowPopover ] = useState( false );
+
+	const getLogo = ( planSlug: string ) => {
```
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/card-content.tsx:28]: Non-blocking:

If we're simply mapping from one value to another, an alternative to a function and a `switch-case` block would be an object:

```js
const LOGOS = {
	[ JETPACK_HOSTING_WPCOM_BUSINESS ]: <CloudLogo />,
	[ JETPACK_HOSTING_WPCOM_ECOMMERCE ]: <WooLogo />,
};

const myLog = LOGOS[ planSlug ]; // automatically returns null if not found
```

Though, on second thought, this approach renders two components every time. 🤷

Comment (yashwin) [reply client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/card-content.tsx:28]: Thanks for your review, @mattgawarecki. This is a good approach. 

This method might change based on the API response. So, I would like to keep this as it is for now. 

> Though, on second thought, this approach renders two components every time. 🤷

Could you please approve this PR if everything else looks good?

**Review by ilonagl — COMMENTED** [review]
Great work, @yashwin! A couple of comments: 

- Would it be possible to reduce the spacing in half between the header + description and price (marked in red)? 

Group 1 (4)

**Comment by ilonagl** [comment]
We discussed that it would be great to reuse on-hover tooltips, as they do in the original plans table. But I don't know how much effort and time that would take. Any thoughts? 


Screenshot 2023-08-31 at 14 50 57

**Comment by ilonagl** [comment]
I wonder why the second product description line appears with a big gap between another description line. Any thoughts?

image

**Comment by yashwin** [comment]
Thanks for the review, @ilonagl! 

> Would it be possible to reduce the spacing in half between the header + description and price (marked in red)?

Yes, I have reduced the spacing here:

mobile (39)
mobile (40)


> We discussed that it would be great to reuse on-hover tooltips, as they do in the original plans table. But I don't know how much effort and time that would take. Any thoughts?

Sure. I have created a ticket here to address this: https://github.com/Automattic/jetpack-avalon/issues/55

> I wonder why the second product description line appears with a big gap between another description line. Any thoughts?

Not sure how this came up in the screenshot. You can check the screenshot I posted above here.

**Comment by ilonagl** [comment]
Thank you, @yashwin! It looks great!

**Review by ilonagl — APPROVED** [review]

File: client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/index.tsx [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+import './style.scss';
 
 export default function WPCOMAtomicHosting() {
 	const translate = useTranslate();
 	const title = translate( 'Create a new WordPress.com site' );
+
+	const plansToBeDisplayed = [
+		'JETPACK_HOSTING_WPCOM_BUSINESS',
+		'JETPACK_HOSTING_WPCOM_ECOMMERCE',
+	]; // Get the plans from the API
```
Comment (vitozev) [inline client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/index.tsx:17]: Do we have the API endpoint built?

**Review by vitozev — APPROVED** [review]
LGTM!

File: client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/index.tsx (hunk 2 above)
Comment (yashwin) [reply client/jetpack-cloud/sections/partner-portal/primary/wpcom-atomic-hosting/index.tsx:17]: I don't think it is ready yet.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8867276

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**